genbadge/utils_badge.py

```diff
@@ -61,33 +61,43 @@ def as_svg(self,
         else:
             # download from requests
             import requests
-            url = 'https://img.shields.io/badge/%s-%s-%s.svg' % (self.left_txt, self.right_txt, self.color)
+            # url encode test
+            safe_left_txt = requests.utils.quote(self.left_txt, safe='')
+            safe_right_txt = requests.utils.quote(self.right_txt, safe='')
+            safe_color_txt = requests.utils.quote(self.color, safe='')         
+            url = 'https://img.shields.io/badge/%s-%s-%s.svg' % (safe_left_txt, safe_right_txt, safe_color_txt)
             response = requests.get(url, stream=True)
             return response.text
 
     def write_to(self,
                  path_or_stream,              # type: Union[TextIO, str, Path]
-                 use_shields=False  # type: bool
+                 use_shields=False,  # type: bool
+                 clear_left_txt=False  # type: bool 
                  ):
         """Write the SVG representation of this badge to the given file
 
         :param path_or_stream:
         :param use_shields:
+        :param clear_left_txt:
         :return:
         """
         # convert to a Path
         if isinstance(path_or_stream, str):
             path_or_stream = Path(path_or_stream)
 
+        svg = self.as_svg(use_shields=use_shields)
+        if clear_left_txt:
+            svg = svg.replace(">" + self.left_txt + "<", "><")
```

Comment:
This is a little messy. In the case where we keep the left side of the badge but have no text it would be great to pass a tab character. However shields escapes label input so this doesn't work well. Instead we're looking for a strict match of the left txt inside html tags and replacing it with nothing afterwards. A placeholder `###` is used to create the svg space (so removing text doesn't remove the left side) in this case. Thoughts on improving this welcome.